datafusion/sql/src/parser.rs

```diff
@@ -381,6 +391,37 @@ impl<'a> DFParser<'a> {
         }
     }
 
+    fn parse_has_options(&mut self) -> bool {
+        self.consume_token(&Token::make_keyword("OPTIONS"))
+    }
+
+    //
+    fn parse_options(&mut self) -> Result<HashMap<String, String>, ParserError> {
+        let mut options: HashMap<String, String> = HashMap::new();
+        if !self.parser.consume_token(&Token::LParen)
+            || self.parser.consume_token(&Token::RParen)
+        {
+            return Ok(options);
+        }
```

Comment:
I think we need to return an error if we see `OPTIONS` that isn't followed by `(`?

```suggestion
        self.parser.expect_token(&Token::LParen)?;
```